### ISSUE TITLE
Network extension - 'OktaHTTPRequestListenerProtocol' is not a member type of 'OktaAuthSdk'

### ISSUE THREAD

**Issue by kochste** [okta/okta-auth-swift#151]
Hi! 

We are looking to inject custom headers with each call to the Okta API by implementing the `OktaHTTPRequestListenerProtocol` protocol and assigning an instance of the implementation to the `httpRequestListener` property of the `OktaApi` type property (`restApi`) within the `OktaAuthStatusUnauthenticated` class instance.  We reference this SDK via the okta-auth-swift cocoapod.

To authenticate our users we call `OktaAuthStatusUnauthenticated.authenticate()`.  

The error we encounter at runtime is "'OktaHTTPRequestListenerProtocol' is not a member type of 'OktaAuthSdk'".

The problem we are encountering appears to be that `OktaHTTPRequestListenerProtocol` is part of the `OktaAuthSdk` module, but there is also an `OktaAuthSdk` class within the module of the same name.

Would it be possible to get a sample of how to use the `OktaHTTPRequestListenerProtocol` properly, via property injection?

Please let me know if I can provide additional information.

Thank you!

**Comment by oleggnidets-okta** [comment]
Hi 
Thanks for the reporting the issue. 
I'm not sure why the error appears. I was able to assign `httpRequestListener` successfully.
`authStatusUnauthenticated.restApi.httpRequestListener = self` where self is View Controller implements `OktaHTTPRequestListenerProtocol`.

If it's possible to provide the example project to reproduce the issue it would be super helpful.

How do you implement `OktaHTTPRequestListenerProtocol`?

**Comment by kochste** [comment]
@oleggnidets-okta thanks for replying to me!  You pointed me in the right direction with your comment.  Once I added an extension to the calling VC, I was able to assign the listener just fine, and the `sendRequest()` function in the extension started to fire.  Prior to that I was trying to keep that assignment out of the VC by creating a new class that implements the `OktaHTTPRequestListenerProtocol` protocol.  I then extended `OktaAPI` and attempted to assign an instance of that listener in the `OktaAPI` extension.  I figured that this might be cleaner as I would not have to do this on each calling VC.

Thanks again!
